zebra-network/src/protocol/external/codec.rs

```diff
@@ -72,6 +75,13 @@ impl Builder {
         }
     }
 
+    /// Configure the codec for the given peer address.
+    #[allow(dead_code)]
+    pub fn for_address(mut self, addr: impl Into<Option<String>>) -> Self {
+        self.addr = addr.into();
```

Comment:
Since this is a builder method, we can assume that anyone who's calling it wants to set the label (otherwise, the could skip calling it), so it can take a `String` rather than an `Option<String>`, which will make the signature a little simpler:

```suggestion
    pub fn with_metrics_label(mut self, metrics_label: String) -> Self {
        self.metrics_label = Some(metrics_label);
```

Reply:
this was my suggestion fwiw, I suggested it because thats approximately how `tracing` does all of its builders, so you could have the default builder default to a mix of on and off states.

Reply:
Thanks, changed to this at https://github.com/ZcashFoundation/zebra/pull/901/commits/16eb2fbc0076c6fa868cb4850c892eb0a31bc952